pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -103,7 +105,14 @@ func handleSyncOp(ctx *downloaderContext, key string,
 		// pass in the config.Name instead of 'filename' which
 		// does not contain the prefix of the relative path with '/'s
 		remoteName = config.Name
-		metricsURL = fmt.Sprintf("S3:%s/%s", dsCtx.Dpath, config.Name)
+		splittedPath := strings.Split(dsPath, "/")
+		if len(splittedPath) > 1 {
+			// if we have '/' in datastore path, move all after first occurrence from bucket to remoteName
+			dsPath = splittedPath[0]
+			splittedPath = append(splittedPath, remoteName)
+			remoteName = strings.Join(splittedPath[1:], "/")
+		}
```

Comment:
When we create path we often use string concatenation with "/" which is ok (but might occasionally create paths with "//" duplication depending on the input. For for taking apart the parth it probably makes sense to use the path package instead of strings.Split to make it be robust against "//" patterns.

Reply:
I put additional logic to remove cases when we have '//' or leading '/'.